layers/dns.go

```diff
@@ -904,6 +959,16 @@ type DNSMX struct {
 	Name       []byte
 }
 
+// DNSOPT is a DNS Option, see RFC6891, section 6.1.2
+type DNSOPT struct {
+	Code uint16
```

Comment:
should this be an enum, maybe "DNSOptionCode"?

Reply:
I can make it an enum and include the values whose status is "Standard" or "Optional" in https://www.iana.org/assignments/dns-parameters/dns-parameters.xhtml#dns-parameters-11

Sounds reasonable?